tests/test_functional.py

```diff
@@ -321,3 +324,50 @@ def test_file_can_have_yaml_references(self):
         assert 'url' in collection['data']['schema']['properties']
         collection = client.get_collection(bucket="main", id="addons")
         assert 'url' in collection['data']['schema']['properties']
+
+
+class KintoWizardTestCase(unittest.TestCase):
+    def setUp(self):
+        self.server = os.getenv("SERVER_URL", "http://localhost:8888/v1")
+        self.auth = os.getenv("AUTH", "user:pass")
+        requests.post(self.server + "/__flush__")
+
+    def load(self, bucket=None, collection=None,
+             filename="tests/dumps/with-schema.yaml", extra=""):
+        return load(self.server, self.auth, filename, bucket, collection, extra)
+
+    def dump(self, bucket=None, collection=None):
+        return dump(self.server, self.auth, bucket, collection)
+
+    def get_client(self):
+        return Client(server_url=self.server, auth=tuple(self.auth.split(':')))
+
+    def test_exit_code_is_one_if_an_error_occured(self):
+        with pytest.raises(exceptions.KintoBatchException):
+            self.load()
+        records = self.get_client().get_records(bucket="natim", collection="toto")
+        assert len(records) == 0
+
+    def test_ignore_4xx_errors_with_parameter(self):
+        # Raises a KintoBatchException in case of error
+        self.load(extra="--ignore-batch-4xx")
+
+    def test_record_updates(self):
+        self.load(extra="--ignore-batch-4xx")
+        client = self.get_client()
+        client.create_record(data={'title': 'titi', 'last_modified': 1496132479110},
+                             id="e2686bac-c45e-4144-9738-edfeb3d9da6d",
+                             collection='toto', bucket='natim')
+        self.load(filename="tests/dumps/with-schema-next.yaml")
+        r = client.get_record(id="e2686bac-c45e-4144-9738-edfeb3d9da6d",
+                              collection='toto', bucket='natim')
+        assert r["data"]["title"] == "toto"
+
+    def test_group_updates(self):
+        self.load(filename="tests/dumps/with-groups.yaml")
+        client = self.get_client()
+        client.update_group(data={"members": ["alexis", "mathieu", "remy"]},
+                            id="toto", bucket="natim")
+        self.load(filename="tests/dumps/with-groups.yaml")
+        r = client.get_group(id="toto", bucket='natim')
+        assert r["data"]["members"] == ["alexis", "mathieu"]
```

Comment:
Should this test be here along tests with `4XX` errors?

Reply:
That's why I made a generic testcase to put all the tests that wheren't directly related to specific import roundtrip. Feel free to add a commit to fix this.